lib/luatest_server.py

```diff
@@ -0,0 +1,110 @@
+    """
+
+    def __init__(self, *args, **kwargs):
+        Test.__init__(self, *args, **kwargs)
+        self.valgrind = kwargs.get('valgrind', False)
+
+    def execute(self, server):
+        """Execute test by luatest command
+
+        Execute 'luatest -v <name>_test.lua -o tap --shuffle none'
+        Provide a verbose output in the tap format.
+        Use shuffle option in none mode for avoiding mixing tests.
+        Use capture mode.
+        """
+        server.current_test = self
+        ts = TestState(self.suite_ini, None, TarantoolServer,
+                       self.run_params,
+                       default_server_no_connect=server)
+        self.inspector.set_parser(ts)
+        command = [os.environ['LUATEST_BIN'], '-v', '-c', self.name,
+                   '-o', 'tap',
+                   '--shuffle', 'none']
+        proc = Popen(command, cwd=server.vardir, stdout=PIPE, stderr=STDOUT)
+        sampler.register_process(proc.pid, self.id, server.name)
+        sys.stdout.write_bytes(proc.communicate()[0])
+
+
+class LuatestServer(Server):
+    """A dummy server implementation for luatest server tests"""
+
+    def __new__(cls, ini=None, *args, **kwargs):
+        cls = Server.get_mixed_class(cls, ini)
+        return object.__new__(cls)
+
+    def __init__(self, _ini=None, test_suite=None):
+        if _ini is None:
+            _ini = {}
+        ini = {'vardir': None}
+        ini.update(_ini)
+        Server.__init__(self, ini, test_suite)
+        self.testdir = os.path.abspath(os.curdir)
+        self.vardir = ini['vardir']
+        self.builddir = ini['builddir']
+        self.name = 'luatest_server'
+
+    @property
+    def logfile(self):
+        return self.current_test.tmp_result
+
+    @property
+    def binary(self):
+        return LuatestServer.prepare_args(self)[0]
+
+    def deploy(self, vardir=None, silent=True, wait=True):
+        self.vardir = vardir
+        if not os.access(self.vardir, os.F_OK):
+            os.makedirs(self.vardir)
+
+    @classmethod
+    def find_exe(cls, builddir):
+        cls.builddir = builddir
+        cls.binary = TarantoolServer.binary
+        cls.debug = bool(re.findall(r'-Debug', str(cls.version()),
+                                    re.I))
+
+    @staticmethod
+    def find_tests(test_suite, suite_path):
+        """Looking for *_test.lua, which are can be executed by luatest."""
+
+        def patterned(test, patterns):
+            answer = []
+            for i in patterns:
+                if test.name.find(i) != -1:
+                    answer.append(test)
+            return answer
+
+        test_suite.ini['suite'] = suite_path
+        tests = glob.glob(os.path.join(suite_path, "*_test.lua"))
+
+        if not tests:
+            executable_path_glob = os.path.join(test_suite.args.builddir,
+                                                'test', suite_path, '*_test')
+            tests = glob.glob(executable_path_glob)
```

Comment:
I doubt that we should find tests in the binary dir. It is actual for unit tests, because we need to compile them. However it is not the case for Lua files.